Migrate to chart dependency for upstream grafana chart

QuantumEnigmaa opened the pull request:
towards https://github.com/giantswarm/giantswarm/issues/24518

:warning: DO NOT MERGE :warning: - Need to prepare the update of config repo's custom values and to synchronize release grafana and config.

QuentinBisson commented:
Can you regerenate the schema?

QuentinBisson commented on .circleci/config.yml:
```diff
@@ -7,24 +7,28 @@ workflows:
           context: "architect"
-          executor: "app-build-suite"
+          # app-build-suite breaks on "application.giantswarm.io/team" in _helpers test
+          #executor: "app-build-suite"
```
I'd like it if we added this tbh

hervenicol replied:
Requires changes in app-build-suite. Today, all our "chart-depending" apps have had to remove this.
I agree though, would be good to update app-build-suite now we know it's the way we're going with our apps.

QuentinBisson replied:
I added it to the prometheus-operator-app without issues though. Why do we need to remove it?

QuentinBisson commented on .circleci/ct-config.yaml:
```diff
@@ -0,0 +1,3 @@
+---
+chart-repos:
+- grafana=https://grafana.github.io/helm-charts
```
```suggestion
- grafana=https://grafana.github.io/helm-charts

```

TheoBrigitte replied:
New line at EOF is missing

QuentinBisson commented on .gitignore:
```diff
@@ -4,3 +4,4 @@ test-results/
 **/__pycache__
 **/test_results**.xml
 build/
+helm/grafana/charts/*.tgz
```
```suggestion
helm/grafana/charts/*.tgz

```

TheoBrigitte replied:
New line at EOF is missing

QuentinBisson commented on helm/grafana/Chart.yaml:
```diff
@@ -1,27 +1,20 @@
 apiVersion: v2
 name: grafana
-version: 1.1.0
+version: [[ .Version ]]
```
```suggestion
version: 2.0.0
```

hervenicol replied:
Current CI fails if we have the wrong `version`, and suggests we set it to `"[[ .Version ]]`.

QuentinBisson replied:
If we use architect yes, with ABS we need to set the version manually

QuantumEnigmaa replied:
I updated the Chart.yaml with this change

QuentinBisson commented:
Can you remove the binary in the charts/ directory?

TheoBrigitte commented on CHANGELOG.md:
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+
+- Changed custom grafana helm chart to use chart dependency for upstream grafana chart. This will allow easier update of grafana version.
+- Custom changes written in _helpers.tpl.
+- Added ct-config file in .circleci folder and updated the config one.
```
This is an internal CI change, I would not mention this in the changelog.

TheoBrigitte commented on .circleci/config.yml:
```diff
@@ -7,24 +7,28 @@ workflows:
       - architect/push-to-app-catalog:
           context: "architect"
-          executor: "app-build-suite"
+          # app-build-suite breaks on "application.giantswarm.io/team" in _helpers test
+          #executor: "app-build-suite"
```
Can we solves the error instead of changing the executor ?

QuentinBisson commented on helm/grafana/templates/_helpers.tpl:
```diff
@@ -74,72 +11,7 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
-  {{- else -}}
-    {{- print "extensions/v1beta1" -}}
-  {{- end -}}
+application.giantswarm.io/team: "atlas"
```
That will solve the abs issue

hervenicol replied:
This would not set the right team label because, in the sub-chart, `application.giantswarm.io/team` does not exist / is empty.

QuentinBisson replied:
This is working for the promethues-operator-app so I;m a bit perplexed

QuentinBisson replied:
Actually it was not :( how did I miss it ..

QuentinBisson replied:
This should solve it https://github.com/giantswarm/app-build-suite/pull/234

QuantumEnigmaa replied:
I updated the label with the annotation you suggested

TheoBrigitte commented on helm/grafana/Chart.yaml:
```diff
@@ -1,27 +1,20 @@
 apiVersion: v2
 name: grafana
-version: 1.1.0
+version: [[ .Version ]]
```
I guess this goes along with changing the executor in the CI. Could we fix the error and keep the version here ?

QuantumEnigmaa replied:
As I mentioned with Quentin, I have now put manually version: 2.0.0 there. Therefore we can keep the version here.

QuentinBisson commented:
Can you post a diff between the old template and the new template usign `helm template helm/grafana` before and after the change?

TheoBrigitte commented on helm/grafana/Chart.yaml:
```diff
@@ -1,27 +1,20 @@
-    email: miroslav.hadzhiev@gmail.com
-  - name: torstenwalter
-    email: mail@torstenwalter.de
+  - name: Giant Swarm applications team
+    url: https://github.com/giantswarm/grafana-app
```
Maintainers is Giantswarm Team Atlas
```suggestion
  - name: giantswarm/team-atlas
    email: team-atlas@giantswarm.io
```

hervenicol replied:
We took it from loki-app IIRC, so we should update it there too :grin:

TheoBrigitte replied:
Here you go https://github.com/giantswarm/loki-app/pull/126

TheoBrigitte reviewed (changes requested):
Nice job. But let's try to fix CI if we can.

hervenicol commented:
Also, added "do not merge" mention in PR description. Better safe than sorry :grin:

QuantumEnigmaa commented:
I updated the values.schema.json file as well as the CHANGELOG and the Chart.yaml, removed the binary from the `charts` folder and added a new line at EOF for each specified file.

QuentinBisson commented on .abs/main.yaml:
```diff
@@ -1,4 +1,6 @@
 generate-metadata: true
+destination: ./build
 catalog-base-url: https://giantswarm.github.io/giantswarm-catalog/
+skip-steps: validate
```
Can you add a comment explaining why? Otherwise looks good to me :)

QuentinBisson replied:
Otherwise LGTM

QuentinBisson reviewed (approved):
LGTM, I'd like some other eyes though :)

TheoBrigitte reviewed (approved):
LGTM, please make sure to test deploy this both fresh install and via upgrade.

hervenicol commented:
Tested on a MC (on `ginger`), using config from https://github.com/giantswarm/config/pull/1406 and app from https://github.com/giantswarm/grafana-app/pull/74.

Looks good:
* grafana deployed
* dex authentication
* dashboards are there
* datasource is OK

QuantumEnigmaa commented:
Tested on a WC (also on `ginger`), using happa app installation tab.

Encountered AppArmor issue mentioned here : https://github.com/giantswarm/giantswarm/issues/24677 
Apart for this, looks good : 
- grafana deployed 
- authentication working

QuentinBisson commented:
Awesome work <3

hervenicol reviewed (approved):
I think now we're good for release :+1:
